view/src/main/java/org/xcolab/view/pages/profile/beans/BadgeBean.java

```diff
@@ -1,82 +1,69 @@
 package org.xcolab.view.pages.profile.beans;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import org.xcolab.client.contest.ContestClientUtil;
-import org.xcolab.client.contest.exceptions.ContestNotFoundException;
-import org.xcolab.client.contest.pojo.Contest;
 import org.xcolab.client.contest.pojo.phases.ContestPhase;
 import org.xcolab.client.contest.pojo.phases.ContestPhaseRibbonType;
-import org.xcolab.client.proposals.ProposalAttributeClientUtil;
 import org.xcolab.client.proposals.ProposalClientUtil;
 import org.xcolab.client.proposals.ProposalPhaseClientUtil;
-import org.xcolab.client.proposals.enums.ProposalAttributeKeys;
 import org.xcolab.client.proposals.pojo.Proposal;
 import org.xcolab.client.proposals.pojo.phases.ProposalContestPhaseAttribute;
 import org.xcolab.util.enums.contest.ProposalContestPhaseAttributeKeys;
 import org.xcolab.view.pages.profile.entity.Badge;
 
 import java.io.Serializable;
 import java.util.ArrayList;
-import java.util.Date;
+import java.util.Comparator;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
 
 public class BadgeBean implements Serializable {
 
     private static final long serialVersionUID = 1L;
-    private final static Logger _log = LoggerFactory.getLogger(BadgeBean.class);
-    private final long userId;
-    private final List<Badge> badges;
+
+    private final List<Badge> badges = new ArrayList<>();
 
     public BadgeBean(long userId) {
-        this.userId = userId;
-        badges = new ArrayList<>();
-        fetchBadges();
+        populateBadges(userId);
```

Comment:
Why not use `badges = getBadges(userId);` here? I think that's more clear than a call to a `void` method where you have to read through the whole method to understand the side effects.

Reply:
Good point, that's much better. Done!